oauth2/oauth2-jose/src/main/java/org/springframework/security/oauth2/jwt/NimbusJwtDecoder.java

```diff
@@ -271,7 +272,16 @@ public static final class JwkSetUriJwtDecoderBuilder {
 
 		private Set<SignatureAlgorithm> signatureAlgorithms = new HashSet<>();
 
-		private RestOperations restOperations = new RestTemplate();
+		private RestOperations restOperations = this.buildRestTemplateWithDefaultTimeoutValues();
+
+		private RestTemplate buildRestTemplateWithDefaultTimeoutValues() {
```

Comment:
For now, will you please isolate your changes to `JwtDecoderProviderConfigurationUtils` since that `RestTemplate` is otherwise not configurable? The reason I say that is that I'd prefer to rely on `RestTemplate` for defaults as a general rule.